dev/reference/best-practices/massive-regions.md

```diff
@@ -0,0 +1,143 @@
+
+### Method 4: Enable Hibernate Region
+
+In the actual situation, read and write requests are not evenly distributed on every Region. Instead, they are concentrated on a few Regions. Then you can minimize the number of messages for the temporarily idle Regions, which is the feature of Hibernate Region. In this feature, Raftstore does sent tick messages to the Raft state machines of idle Regions when necessary. Then these Raft state machines will not be triggered to generate heartbeat information, greatly reducing the workload of Raftstore.
+
+Up to TiDB v3.0.5, Hibernate Region is still an experimental feature, which is enabled by default in [TiKV master](https://github.com/tikv/tikv/tree/master). You can enable this feature according to your actual need. For the configuration of Hibernate Region, refer to [Configure Hibernate Region](https://github.com/tikv/tikv/blob/master/docs/reference/configuration/raftstore-config.md#hibernate-region).
+
+### Method 5: Enable `Region Merge`
+
+> **Note:**
+>
+> `Region Merge` is enabled in TiDB v3.0 by default.
+
+You can also reduce the number of Regions by enabling `Region Merge`. Contrary to `Region Split`, `Region Merge` is the process of merging adjacent small Regions through scheduling. After dropping data or executing the `Drop Table` or `Truncate Table` statement, you can merge small Regions or even empty Regions to reduce resource consumption.
+
+Enable `Region Merge` by setting the following configurations:
+
+{{< copyable "" >}}
+
+```
+>> pd-ctl config set max-merge-region-size 20
+>> pd-ctl config set max-merge-region-keys 200000
+>> pd-ctl config set merge-schedule-limit 8
+```
+
+Refer to [Region Merge](https://github.com/tikv/tikv/blob/master/docs/how-to/configure/region-merge.md) and [PD configuration file](/dev/reference/configuration/pd-server/configuration-file.md#schedule) for more details.
+
+The default configuration of the `Region Merge` parameters is rather conservative. You can speed up the `Region Merge` process by referring to the method provided in [PD Scheduling Best Practices](/dev/reference/best-practices/pd-scheduling.md#region-merge-is-slow).
+
+## Other problems and solutions
+
+### Switching PD Leader is slow
+
+PD needs to persist Region Meta information on etcd to ensure that PD can quickly resume to provide Region routing services after switching the PD Leader node. As the number of Regions increases, the performance problem of etcd appears, making it slower for PD to get Region Meta information from etcd when PD is switching the Leader. With millions of Regions, it might take dozens or even tens of seconds to get the meta information from etcd.
+
+To address this problem, `use-region-storage` is enabled by default in PD since TiDB v3.0. With this feature enabled, PD stores Region Meta information on local LevelDB and synchronizes the information among PD nodes through other mechanism.
```

Comment:
Please confirm whether the **"synchronize"** is correctly used here. Is it just "obtain" or another meaning?

```suggestion
To address this problem, `use-region-storage` is enabled by default in PD in TiDB v3.0. With this feature enabled, PD stores Region Meta information on local LevelDB and synchronizes the information among PD nodes through other mechanisms.
```

Reply:
**"synchronize"** is correct here, as confirmed with the author.